cloud/jenkins/pxco-openshift.groovy

```diff
@@ -0,0 +1,540 @@
+            defaultValue: 'https://github.com/percona/percona-xtradb-cluster-operator',
+            description: 'percona-xtradb-cluster-operator repository',
+            name: 'GIT_REPO')
+        string(
+            defaultValue: 'latest',
+            description: 'OpenShift kubernetes version. If set to min or max, value will be automatically taken from release_versions file.',
+            name: 'PLATFORM_VER')
+        choice(
+            choices: 'YES\nNO',
+            description: 'Run tests in cluster wide mode',
+            name: 'CLUSTER_WIDE')
+        string(
+            defaultValue: '',
+            description: 'Operator image: perconalab/percona-xtradb-cluster-operator:main',
+            name: 'IMAGE_OPERATOR')
+        string(
+            defaultValue: '',
+            description: 'PXC image: perconalab/percona-xtradb-cluster-operator:main-pxc8.0',
+            name: 'IMAGE_PXC')
+        string(
+            defaultValue: '',
+            description: 'PXC proxy image: perconalab/percona-xtradb-cluster-operator:main-proxysql',
+            name: 'IMAGE_PROXY')
+        string(
+            defaultValue: '',
+            description: 'PXC haproxy image: perconalab/percona-xtradb-cluster-operator:main-haproxy',
+            name: 'IMAGE_HAPROXY')
+        string(
+            defaultValue: '',
+            description: 'Backup image: perconalab/percona-xtradb-cluster-operator:main-pxc8.0-backup',
+            name: 'IMAGE_BACKUP')
+        string(
+            defaultValue: '',
+            description: 'PXC logcollector image: perconalab/percona-xtradb-cluster-operator:main-logcollector',
+            name: 'IMAGE_LOGCOLLECTOR')
+        string(
+            defaultValue: '',
+            description: 'PMM client image: perconalab/pmm-client:dev-latest',
+            name: 'IMAGE_PMM_CLIENT')
+        string(
+            defaultValue: '',
+            description: 'PMM server image: perconalab/pmm-server:dev-latest',
+            name: 'IMAGE_PMM_SERVER')
+    }
+    agent {
+        label 'docker'
+    }
+    options {
+        buildDiscarder(logRotator(daysToKeepStr: '-1', artifactDaysToKeepStr: '-1', numToKeepStr: '30', artifactNumToKeepStr: '30'))
+        skipDefaultCheckout()
+        disableConcurrentBuilds()
+        copyArtifactPermission('pxc-operator-latest-scheduler');
+    }
+    stages {
+        stage('Prepare Node') {
+            steps {
+                prepareNode()
+            }
+        }
+        stage('Docker Build and Push') {
+            steps {
+                dockerBuildPush()
+            }
+        }
+        stage('Init Tests') {
+            steps {
+                initTests()
+            }
+        }
+        stage('Run Tests') {
+            parallel {
+                stage('cluster1') {
+                    options {
+                        timeout(time: 3, unit: 'HOURS')
+                    }
+                    steps {
+                        clusterRunner('c1')
```

Comment:
What for do we change cluster1 to c1?

Reply:
When running `xxx-openshift-latest` Jenkins pipelines, I encountered a warning:
`level=warning msg=Length of cluster name "openshift-lat-pxc-jenkins-cluster1" is 34 which is greater than the max 21 allowed. The name will be truncated to "openshift-lat-pxc-jen".`
This warning appears randomly, even when the cluster name is too long, such as `openshift-lat-pxc-jenkins-cluster1`. The random nature of the warning complicates debugging and, to save time and lean on the safe side, I truncated cluster name and left the investigation of the root cause for later.
To avoid potential issues, I modified the naming convention for Openshift cluster names:
**Current format:**

- jenkins-$JOB_NAME-$GIT_SHORT_COMMIT-$CLUSTER_SUFFIX
- Example: openshift-lat-pxc-jenkins-cluster1 (34 characters)
**New format:**

- $JOB_NAME-$GIT_SHORT_COMMIT-$CLUSTER_SUFFIX
- Example: pxco-os-1-e9cb9a8f-c1 (21 characters)
**Additional changes:**
Renamed the job from `pxco-openshift-01` to `pxco-os-1`.
Shortened `$CLUSTER_SUFFIX` from `cluster1`, `cluster2`, etc., to `c1`, `c2`, etc.
These adjustments ensure the cluster name stays within the 21-character limit, eliminating the warning and reducing the risk of naming conflicts.
Note: I won’t investigate the root cause of the warning further at this time, as it does not consistently appear and does not directly affect functionality.